include/zfs_namecheck.h

```diff
@@ -43,6 +43,8 @@ typedef enum {
 	NAME_ERR_RESERVED,		/* entire name is reserved */
 	NAME_ERR_DISKLIKE,		/* reserved disk name (c[0-9].*) */
 	NAME_ERR_TOOLONG,		/* name is too long */
+	NAME_ERR_SELF_REFERENCE,	/* self path reference in filesystem*/
+	NAME_ERR_PARENT_REFERENCE,	/* parent path reference in filesystem*/
```

Comment:
nit: Like the `NAME_ERR_DISKLIKE` case (and printed error messages) I think it would be very helpful to include the '.', and ',,' strings in these comments.  What do you think about the following.

```c
/* reserved self path name ('.') */
/* reserved parent path name ('..') */
```

Reply:
OK